vochain/account.go

```diff
@@ -461,250 +473,253 @@ func (v *State) SetAccount(accountAddress common.Address, account *Account) erro
 	if err != nil {
 		return err
 	}
-	v.Tx.Lock()
-	defer v.Tx.Unlock()
-	log.Debugf("setAccount: {address %s, nonce %d, infoURI %s, balance: %d, delegates: %+v}",
+	log.Debugf(`setAccount: {
+			address %s,
+			nonce %d,
+			infoURI %s,
+			balance: %d,
+			delegates: %+v,
+			processIndex: %d
+		}`,
 		accountAddress.String(),
 		account.Nonce,
 		account.InfoURI,
 		account.Balance,
 		printPrettierDelegates(account.DelegateAddrs),
+		account.ProcessIndex,
 	)
+	for _, l := range v.eventListeners {
+		l.OnSetAccount(accountAddress.Bytes(), account)
```

Comment:
I think passing a the account itself makes more sense than just the bytes of the address. But take care on not passing it as a pointer but as a value.

Reply:
already passing the account, but changed to pass a copy